src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -567,16 +566,25 @@ def specify_all_power_straps_by_tracks(self, layer_names: List[str], ground_net:
             offset = layer.offset # TODO this is relaxable if we can auto-recalculate this based on hierarchical setting
 
             add_pins = layer_name in pin_layers
-            nets = [ground_net, power_nets[0]] # TODO this needs to change when implementing multiple power domains
-            output.extend(self.specify_power_straps_by_tracks(layer_name, last.name, blockage_spacing, track_pitch, track_width, track_spacing, track_start, offset, bbox, nets, add_pins))
+            # For multiple domains, we'll stripe them like this:
+            # 2:1 :   A A B A A B ...
+            # 3:1 :   A A A B A A A B ...
+            # 3:2 :   A A A B B A A A B B ...
+            # 2:2:1 : A A B B C A A B B C ...
+            sum_weights = sum(power_weights)
+            for i in range(sum_weights):
+                nets = [ground_net, power_nets[i]]
+                group_offset = track_start + track_pitch * i
+                group_pitch = sum_weights * track_pitch
+                output.extend(self.specify_power_straps_by_tracks(layer_name, last.name, blockage_spacing, group_pitch, track_width, track_spacing, group_offset, offset, bbox, nets, add_pins))
             last = layer
         return output
 
     _power_straps_last_index = -1
 
     def _power_straps_check_index(self, layer_name: str) -> None:
         next_index = self.get_stackup().get_metal(layer_name).index
-        assert next_index > self._power_straps_last_index, "Must construct power straps from bottom to top"
+        assert next_index >= self._power_straps_last_index, "Must construct power straps from bottom to top"
```

Comment:
Is this implying its ok to redo straps from last layer?

Reply:
Yes, this is necessary for the multi-VDD API. I think it's safe... but I could be convinced otherwise. Not sure how to do the multi-VDD without it though.